src/Features/Core/Portable/Shared/Extensions/SyntaxTokenListExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Collections;
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class SyntaxTokenListExtensions
+    {
+        /// <summary>
+        /// Gets the concatenated value text for the token list.
+        /// </summary>
+        /// <returns>The concatenated value text, or an empty string if there are no tokens in the list.</returns>
+        internal static string GetValueText(this SyntaxTokenList tokens)
+        {
+            switch (tokens.Count)
+            {
+                case 0:
+                    return string.Empty;
+
+                case 1:
+                    return tokens[0].ValueText;
```

Comment:
.ValueText means that you're converting certain values if you run into them.  For example, the ValueText for 0.000 will be the same as for 0.0.  Is that what you want?  Or do you want the actual text written into the document?  If so, we'd want .ToString().  But i'm not sure what this is trying to accomplish.